### DISCUSSION TITLE
Allow to run a new workflow with workflow_dispatch created in a pull request

### DISCUSSION THREAD

**Discussion by flodolo** [community/community#21134] [Actions]
If a pull request adds a new workflow which includes `workflow_dispatch`, it should be possible to manually run that action for testing purposes. 

Right now, if an action only runs on schedule or is manually triggered, there's no way to test it from a pull request, you need to temporarily add `on.pull_request`.

**Comment by jsoref** [comment]
If a pull request adds code, you should really pull it into a fork and experiment with it. - This can include adding an additional commit that enables you to run it on push (that commit will run it instantly).

Running a workflow from a fork is incredibly dangerous and making it easy to do this is a bad idea.

**Reply by flodolo** [reply]
I see your point, but I don't think the solution needs to be all or nothing. We're talking about a specific case (workflows that can be manually triggered), and the person who actually run them would be easily limited to the project maintainers.
